pkg/binarywrapper/binarywrapper.go

```diff
@@ -0,0 +1,185 @@
+// Package binarywrapper is a utility package that makes wrapping binaries a little easier
+// as it aims to provide a common structure to use for converting binaries
+// into plugins. Along the way it allows for some setup tasks, validation,
+// and execution.
+package binarywrapper
```

Comment:
can definitely see this as part of something like `vela-plugin-utils-go` like `version`